numpy/core/src/umath/simd.inc.src

```diff
@@ -34,6 +34,22 @@
 
 #define VECTOR_SIZE_BYTES 16
 
+/*
+ * MAX_STEP_SIZE is used to determine if we need to use SIMD version of the ufunc.
+ * Very large step size can be as slow as processing it using scalar. The
+ * value of 2097152 ( = 2MB) was chosen using 2 considerations:
+ * 1) Typical linux kernel page size is 4Kb, but sometimes it could also be 2MB
+      which is == 2097152 Bytes. For a step size as large as this, surely all
+      the loads/stores of gather/scatter instructions falls on 16 different pages
+      which one would think is the worst performance scenario for gather/scatter
+      instructions.
```

Comment:
Is your claim that the performance of using AVX-512 is worse than not using it for this case?